### PR TITLE
#CP-13 feat: assign bus to routes

### PR DESCRIPTION
**Pull request by feyton**
### What the PR does
Allow operators to assign buses to given routes.
As **an operator**, I want to be able to assign a bus to a route so that the bus can only be utilized in that route.
### Description of Task to be completed

- [x] The operator should be able to assign a bus to a route.
- [x] The operator should be able to see a paginated list of buses to routes assignments.
- [x] Appropriate error messages are displayed on unsuccessful attempts.

### How should this be manually tested?
To test this project **(Assuming you already have prior settings, there is no configuration required to test this)**
```
git clone https://github.com/atlp-rwanda/phantom-be-codebandits
cd phantom-be-codebandits
git fetch origin
git checkout ft-assign-bus-to-routes-cp-13
npm install
npm run dev
// Ensure that you have a database up and running. If you prefer to setup on the fly using docker. Refer to the next command before you start the server.

// docker postgress
docker run --name dev-db -e POSTGRES_PASSWORD=1234 -e POSTGRES_USER=phantom -p 6000:5432 -d postgres
// put all necessary configuration in the .env file or rename the .env.sample file
```
After running the server. Test using **Postman** :
[Run in Postman](https://app.getpostman.com/run-collection/19126547-acffda3c-77af-4761-91ef-e4aba6a0257e?action=collection%2Ffork&collection-url=entityId%3D19126547-acffda3c-77af-4761-91ef-e4aba6a0257e%26entityType%3Dcollection%26workspaceId%3D252017b1-dc09-4cf8-ad20-ea23c74ad4d7#?env%5BTEST_ENV%5D=W3sia2V5Ijoie3tiYXNlVXJsfX0iLCJ2YWx1ZSI6Imh0dHA6Ly9sb2NhbGhvc3Q6NTAwMCIsImVuYWJsZWQiOnRydWUsInR5cGUiOiJkZWZhdWx0Iiwic2Vzc2lvblZhbHVlIjoiaHR0cDovL2xvY2FsaG9zdDo1MDAwIiwic2Vzc2lvbkluZGV4IjowfSx7ImtleSI6Int7YWNjZXNzVG9rZW59fSIsInZhbHVlIjoiMTIzNDUiLCJlbmFibGVkIjp0cnVlLCJ0eXBlIjoiZGVmYXVsdCIsInNlc3Npb25WYWx1ZSI6IjEyMzQ1Iiwic2Vzc2lvbkluZGV4IjoxfV0=)
or using **Swagger docs** by heading to [http://localhost:5000](http://localhost:5000) 
### Any background context you want to provide?
Request parameters are mostly sent with request queries. Unlike **parameters**, you can chain more queries to refine your request which is an added advantage when handled customized requests like these ones that take multiple options.
### What are the relevant Trello stories?
- [#CP-13](https://trello.com/c/jRUDZr5t)

### CODE REVIEW

**Comment by iradukundairenee** [comment]
> ### What the PR does
> Allow operators to assign buses to given routes. As **an operator**, I want to be able to assign a bus to a route so that the bus can only be utilized in that route.
> 
> ### Description of Task to be completed
> * [ ]  The operator should be able to assign a bus to a route.
> * [ ]  The operator should be able to see a paginated list of buses to routes assignments.
> * [ ]  Appropriate error messages are displayed on unsuccessful attempts.
> 
> ### How should this be manually tested?
> To test this project **(Assuming you already have prior settings, there is no configuration required to test this)**
> 
> ```
> git clone https://github.com/atlp-rwanda/phantom-be-codebandits
> cd phantom-be-codebandits
> git fetch origin
> git checkout ft-assign-bus-to-routes-cp-13
> npm install
> npm run dev
> // Ensure that you have a database up and running. If you prefer to setup on the fly using docker. Refer to the next command before you start the server.
> 
> // docker postgress
> docker run --name dev-db -e POSTGRES_PASSWORD=1234 -e POSTGRES_USER=phantom -p 6000:5432 -d postgres
> // put all necessary configuration in the .env file or rename the .env.sample file
> ```
> 
> After running the server. Test using **Postman** : [Run in Postman](https://app.getpostman.com/run-collection/19126547-acffda3c-77af-4761-91ef-e4aba6a0257e?action=collection%2Ffork&collection-url=entityId%3D19126547-acffda3c-77af-4761-91ef-e4aba6a0257e%26entityType%3Dcollection%26workspaceId%3D252017b1-dc09-4cf8-ad20-ea23c74ad4d7#?env%5BTEST_ENV%5D=W3sia2V5Ijoie3tiYXNlVXJsfX0iLCJ2YWx1ZSI6Imh0dHA6Ly9sb2NhbGhvc3Q6NTAwMCIsImVuYWJsZWQiOnRydWUsInR5cGUiOiJkZWZhdWx0Iiwic2Vzc2lvblZhbHVlIjoiaHR0cDovL2xvY2FsaG9zdDo1MDAwIiwic2Vzc2lvbkluZGV4IjowfSx7ImtleSI6Int7YWNjZXNzVG9rZW59fSIsInZhbHVlIjoiMTIzNDUiLCJlbmFibGVkIjp0cnVlLCJ0eXBlIjoiZGVmYXVsdCIsInNlc3Npb25WYWx1ZSI6IjEyMzQ1Iiwic2Vzc2lvbkluZGV4IjoxfV0=) or using **Swagger docs** by heading to http://localhost:5000
> 
> ### Any background context you want to provide?
> Request parameters are mostly sent with request queries. Unlike **parameters**, you can chain more queries to refine your request which is an added advantage when handled customized requests like these ones that take multiple options.
> 
> ### What are the relevant Trello stories?
> * [#CP-13](https://trello.com/c/jRUDZr5t)

**Review by renepromesse — CHANGES_REQUESTED** [review]
good work @Yvad60 , could you add screenshots of your requests and responses as well as the swagger docs picture

also see how you can improve on this codeclimate issues

File: src/bustoRouteApp/controllers.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import handleResponse from '../controllers/handleResponse.js';
+import { Bus, Route } from './models.js';
+
+export const createBus = async (req, res) => {
+	const body = matchedData(req, { optional: true });
+	const data = await Bus.get_or_create(body);
+	return handleResponse(res, 201, data);
+};
+
+export const createRoute = async (req, res) => {
```
Comment (samyshyaka) [inline src/bustoRouteApp/controllers.js]: Here, you can create one service and use it on both bus and route creation to avoid duplicating the code

**Review by samyshyaka — DISMISSED** [review]
Nice work @feyton, attend to the feedack below

**Review by renepromesse — DISMISSED** [review]
Hey @feyton , could you fix the coveralls issue?

**Review by samyshyaka — APPROVED** [review]
@feyton, This is a good implementation of the feature and the requested changes have been made. I think the PR is ready to be merged.

**Review by Abdulkeza — APPROVED** [review]
Great Job Mr fab! working fine

**Review by renepromesse — APPROVED** [review]

**Pull request merged**